src/managers/OnOfframpManager.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {IAuth} from "src/misc/interfaces/IAuth.sol";
+import {IERC20} from "src/misc/interfaces/IERC20.sol";
+import {CastLib} from "src/misc/libraries/CastLib.sol";
+import {MathLib} from "src/misc/libraries/MathLib.sol";
+import {IERC165} from "src/misc/interfaces/IERC165.sol";
+import {SafeTransferLib} from "src/misc/libraries/SafeTransferLib.sol";
+
+import {PoolId} from "src/common/types/PoolId.sol";
+import {ShareClassId} from "src/common/types/ShareClassId.sol";
+import {UpdateContractType, UpdateContractMessageLib} from "src/spoke/libraries/UpdateContractMessageLib.sol";
+
+import {IBalanceSheet} from "src/spoke/interfaces/IBalanceSheet.sol";
+import {IUpdateContract} from "src/spoke/interfaces/IUpdateContract.sol";
+
+import {IOnOfframpManager} from "src/managers/interfaces/IOnOfframpManager.sol";
+import {IOnOfframpManagerFactory} from "src/managers/interfaces/IOnOfframpManagerFactory.sol";
+import {IDepositManager, IWithdrawManager} from "src/managers/interfaces/IBalanceSheetManager.sol";
+
+contract OnOfframpManager is IOnOfframpManager {
+    using CastLib for *;
+    using MathLib for uint256;
+
+    PoolId public immutable poolId;
+    address public immutable spoke;
+    ShareClassId public immutable scId;
+    IBalanceSheet public immutable balanceSheet;
+
+    bool public permissionlessOnramp;
+    bool public permissionlessOfframp;
+    mapping(address => bool) public manager;
```

Comment:
Nobody sets this storage. Why not use the manager from the BSM?

Reply:
It should be a separate permission layer. This is indeed missing a setter, and we should possibly rename this